### PR TITLE
ci: add distroless image

### PR DESCRIPTION
**Pull request by TartanLeGrand**
Add distroless image for size, use non-root user and add ide to gitignore.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,31 +1,23 @@
...
-FROM build_deps AS build
-
 COPY . .
 
-RUN CGO_ENABLED=0 go build -o webhook -ldflags '-s -w -extldflags "-static"' .
+RUN go mod download
+RUN CGO_ENABLED=0 go build -o /go/bin/app -ldflags '-s -w -extldflags "-static"' .
 
-FROM alpine:3.16
+FROM gcr.io/distroless/base-debian11
```
Comment (aureq) [inline Dockerfile:12]: Could you please point out the benefits of using a distroless base image?

**Comment by aureq** [comment]
@TartanLeGrand have you seen my comments above ?

**Comment by TartanLeGrand** [comment]
@aureq I have no comments from you in this PR. Have you been send your review ?

**Review by aureq — CHANGES_REQUESTED** [review]

**Comment by aureq** [comment]
> @aureq I have no comments from you in this PR. Have you been send your review ?

I did not 🤦‍♂️ though you should be able to see them now.

File: Dockerfile [hunk 2]
```diff
@@ -1,31 +1,23 @@
...
 
-RUN apk upgrade --no-cache && \
-    apk add --no-cache ca-certificates libcap
+COPY --from=build /go/bin/app /
 
-COPY --from=build /workspace/webhook /usr/local/bin/webhook
+COPY --from=build /sbin/getcap /sbin
```
[the remark this answers is not in the document]
Comment (TartanLeGrand) [reply Dockerfile:16]: Mistake for debuging.

File: Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (TartanLeGrand) [reply Dockerfile:12]: Size and security vs alpine image

File: Dockerfile [hunk 3]
```diff
@@ -1,31 +1,23 @@
...
+COPY --from=build /sbin/setcap /sbin
+COPY --from=build /lib/*-linux-gnu/libcap.so.2 /lib
 
-# allow bind() for ports < 1024 as non-root
-RUN setcap cap_net_bind_service=+ep /usr/local/bin/webhook
+RUN ["/sbin/setcap", "cap_net_bind_service=+ep", "/app"]
```
[the remark this answers is not in the document]
Comment (TartanLeGrand) [reply Dockerfile:20]: it doesn't change anything to the execution of the command. It is just not put in the /usr/local/bin folder and that the command is not called webhook and that I used a more generic term.

File: Dockerfile [hunk 4]
```diff
@@ -1,31 +1,23 @@
...
 
-ENTRYPOINT ["/usr/local/bin/webhook"]
+USER nonroot:nonroot
```
[the remark this answers is not in the document]
Comment (TartanLeGrand) [reply Dockerfile:22]: The `nonroot` user is the user created in the distroless image, this is to match the distroless convention.

**Comment by aureq** [comment]
I've partially merged your PR, thank you for this contribution.

For the purpose of transparency, the base images will remain on Alpine Linux for the following reasons:

- `gcr.io/distroless/base-debian11:latest` is 20.3MB and has 13 reported [vulnerabilities](https://trivy.dev/results/?image=gcr.io/distroless/base-debian11)
- `alpine:3.16` is 5.54MB and has no reported [vulnerabilities](https://trivy.dev/results/?image=alpine:3.16)

**Pull request merged**